packages/hop-node/src/watchers/classes/L2Bridge.ts

```diff
@@ -138,8 +140,12 @@ export default class L2Bridge extends Bridge {
     return await this.mapEventsBatch(this.getTransferSentEvents, cb, options)
   }
 
-  async getTransferSentTimestamp (transferId: string): Promise<number> {
+  async getTransferSentEvent (transferId: string): Promise<TransferSentEvent | null> {
```

Comment:
IIUC this limits `getTransferSentEvent` to only look back 7 days. I would expect a function called `getTransferSentEvent` to look back as far as needed in order to get the passed-in `transferId`. What is the purpose of limiting this function to a 7 day lookback?

Reply:
The reason was because it was searching only like 3 days back on optimism without explicitly specifying the start block number on certain transfer Ids. I'm not sure if that was expected or if there's a bug in eventsBatch. Since this method  is used for filling only incomplete items, the 7 days was chosen to minimize rpc lookups because we don't do anything with transfers older than 7 days. I don't really like it either so I'm good with any suggestions